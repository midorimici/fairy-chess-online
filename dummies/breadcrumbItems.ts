export const items = [
  {
    name: 'Home',
    path: '/',
  },
  {
    name: 'Docs',
    path: '/docs',
  },
  {
    name: 'Breadcrumb',
    path: '/docs/breadcrumb',
  },
];
