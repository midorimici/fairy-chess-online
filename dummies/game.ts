export const game = {
  id: '1234567890abcdef',
  name: 'Game name',
  description:
    'Game desciption. This part may contains the description about the game or/and pieces used in the game.',
  initialBoardImage: {
    url: 'https://media.graphcms.com/zzxpvUPeT1yHQAO8vSrI',
    width: 256,
    height: 256,
  },
};
