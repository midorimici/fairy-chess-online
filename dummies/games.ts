export const games = [
  {
    id: '1234567890abcdef',
    name: 'First game',
    description:
      'Game desciption. This part may contains the description about the game or/and pieces used in the game.',
    initialBoardImage: {
      url: 'https://media.graphcms.com/zzxpvUPeT1yHQAO8vSrI',
      width: 256,
      height: 256,
    },
  },
  {
    id: '1234567890abcde0',
    name: 'Second game',
    description:
      'Game desciption. This part may contains the description about the game or/and pieces used in the game.',
    initialBoardImage: {
      url: 'https://media.graphcms.com/zzxpvUPeT1yHQAO8vSrI',
      width: 256,
      height: 256,
    },
  },
];
