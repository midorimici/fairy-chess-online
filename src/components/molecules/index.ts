export { BreadcrumbNav } from './BreadcrumbNav';
export { Container } from './Container';
export { GameList } from './GameList';
