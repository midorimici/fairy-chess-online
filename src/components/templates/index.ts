export { Game } from './Game';
export { Games } from './Games';
export { Home } from './Home';
export { NewRoom } from './NewRoom';
