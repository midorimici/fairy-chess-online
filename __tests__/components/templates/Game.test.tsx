import { render, screen } from '@testing-library/react';
import { game } from 'dummies/game';
import { Game } from '~/components/templates';

describe('Game', () => {
  it('renders unchanged', () => {
    const { asFragment } = render(<Game game={game} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('contains copy right', () => {
    render(<Game game={game} />);
    const copyRightElement = screen.getByText(/Â©/);
    expect(copyRightElement).toBeInTheDocument();
  });
});
