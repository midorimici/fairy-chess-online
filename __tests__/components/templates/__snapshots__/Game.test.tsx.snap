// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders unchanged 1`] = `
<DocumentFragment>
  <div
    class="chakra-container css-1y4v3tz"
  >
    <main
      class="css-i9gxme"
    >
      <div
        class="chakra-stack css-p27qbx"
      >
        <nav
          aria-label="breadcrumb"
          class="chakra-breadcrumb css-0"
        >
          <ol
            class="chakra-breadcrumb__list css-0"
          >
            <li
              class="chakra-breadcrumb__list-item css-vn2883"
            >
              <a
                class="chakra-breadcrumb__link css-0"
                href="/games"
              >
                games
              </a>
              <span
                class="css-t4q1nq"
                role="presentation"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ChevronRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </span>
            </li>
            <li
              class="chakra-breadcrumb__list-item css-vn2883"
            >
              <span
                aria-current="page"
                class="chakra-breadcrumb__link css-0"
              >
                Game name
              </span>
            </li>
          </ol>
        </nav>
        <div
          class="chakra-stack css-snbr68"
        >
          <div
            class="chakra-stack css-1vpxlgj"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjU2IiBoZWlnaHQ9IjI1NiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Game name"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="chakra-text css-1n49pl7"
            >
              Game name
            </p>
            <p
              class="chakra-text css-0"
            >
              Game desciption. This part may contains the description about the game or/and pieces used in the game.
            </p>
          </div>
          <div
            class="chakra-stack css-p27qbx"
          >
            <button
              class="chakra-button css-5c2176"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LanguageIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
                  />
                </svg>
              </span>
              play-public
            </button>
            <button
              class="chakra-button css-5c2176"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LockIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                  />
                </svg>
              </span>
              play-private
            </button>
          </div>
        </div>
      </div>
    </main>
    <div
      class="css-gmuwbf"
    >
      Â© 2022-22 midorimici
    </div>
  </div>
</DocumentFragment>
`;
